ibis/pandas/core.py

```diff
@@ -324,6 +388,10 @@ def main_execute(expr, params=None, scope=None, aggcontext=None, **kwargs):
     if scope is None:
         scope = {}
 
+    if timecontext is not None:
+        # convert timecontext to datetime type, if time strings are provided
+        timecontext = canonicalize_context(timecontext)
```

Comment:
this is the very top level rigth?

Reply:
Yes this is the very top level of execution.